src/solvers/smt2_incremental/smt_response_validation.cpp

```diff
@@ -37,5 +37,14 @@ template class response_or_errort<smt_responset>;
 
 response_or_errort<smt_responset> validate_smt_response(const irept &parse_tree)
 {
+  if(parse_tree.id() == "sat")
```

Comment:
⛏️ Why not `switch (parse_tree.id()) { case "sat": }`. Might make it a bit more readable?

Reply:
Because you can't switch on a string in C++.

Reply:
😱

Reply:
But you can switch on a/the first character... so if you want a big nested tree of case statements... ;)